### PR TITLE
[create-sitecore-jss] Allow proxies to be initialized with main app and rework angular initializer

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
* Introduces new `nodeAppDestination` arg for `create-sitecore-jss`. It will be used as destination for node proxy apps, when present
* Allows proxy apps to be installed alongside main apps (devex win!)
* Adds some initial meat to `angular-xmcloud` and `node-xmcloud-proxy` templates, simplifying smoke testing

## Testing Details
- [x] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/bin.ts [hunk 1]
```diff
@@ -26,6 +36,60 @@ export const parseArgs = (): ParsedArgs => {
...
+  }
+  // validate/gather destination
+  const defaultBaseDestination = `${process.cwd()}${
+    args.appName ? sep + args.appName : `${sep}${templates[0]}`
+  }`;
+  const destination = args.destination
+    ? args.destination
+    : args.yes
+    ? defaultBaseDestination
+    : await promptDestination('Where would you like your new app created?', defaultBaseDestination);
```
Comment (sc-addypathania) [inline packages/create-sitecore-jss/src/bin.ts:51]: Should we use regular if/else for the ease of reading this?

File: packages/create-sitecore-jss/src/bin.ts [hunk 2]
```diff
@@ -26,6 +36,60 @@ export const parseArgs = (): ParsedArgs => {
...
+  if (proxyApp) {
+    const defaultProxyDestination = proxyApp && `${process.cwd()}${sep}${proxyApp}`;
+    let nodeAppDestination = args.nodeAppDestination
+      ? args.nodeAppDestination
+      : args.yes
+      ? defaultProxyDestination
+      : await promptDestination(
+          'Where would you like your proxy app created?',
+          defaultProxyDestination
+        );
```
Comment (sc-addypathania) [inline packages/create-sitecore-jss/src/bin.ts:64]: same comment as above

File: packages/create-sitecore-jss/src/bin.ts (hunk 1 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/bin.ts:51]: I can switch to if/else. The way it's done now was to not have too large of a duplicating code block (same logic is used further down)

File: packages/create-sitecore-jss/src/bin.ts [hunk 3]
```diff
@@ -6,13 +6,23 @@ import { initRunner } from './init-runner';
...
   // parse any command line arguments passed into `init sitecore-jss`
   // to pass to the generator prompts and skip them.
   // useful for CI and testing purposes
   const options = {
     boolean: ['appPrefix', 'force', 'noInstall', 'yes', 'silent', 'prePushHook'],
-    string: ['appName', 'destination', 'templates', 'hostName', 'fetchWith', 'language'],
+    string: [
+      'appName',
+      'destination',
+      'nodeAppDestination',
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/bin.ts:20]: I think we should change this to `proxyDestination` for the CLI param for clarity and consistency with prompts (and technically even the main app is a node app).

File: packages/create-sitecore-jss/src/common/args/base.ts [hunk 4]
```diff
@@ -33,6 +33,10 @@ export type BaseArgs = {
    * Pre-push hook for linting check
    */
   prePushHook?: boolean;
+  /**
+   * Optional destination for proxy app, whether it's initialized alongside the main one or standalone
+   */
+  nodeAppDestination?: string;
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/common/args/base.ts:39]: Same comment here (rename to `proxyDestination`)

File: packages/create-sitecore-jss/src/init-runner.ts [hunk 5]
```diff
@@ -41,16 +41,19 @@ export const initRunner = async (initializers: string[], args: BaseArgs) => {
...
 
-  saveConfiguration(args.templates, path.resolve(`${args.destination}${sep}package.json`));
-  // final steps (install, lint, etc)
-  if (!args.noInstall) {
-    installPackages(args.destination, args.silent);
-    lintFix(args.destination, args.silent);
-  }
+  for (const destination of [args.destination, args.nodeAppDestination]) {
+    if (!destination) continue;
+    saveConfiguration(args.templates, path.resolve(`${destination}${sep}package.json`));
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/init-runner.ts:46]: Not sure if saving the applied "templates" makes sense for the proxy app. Should we save this saveConfiguration operation only for the primary app?

File: packages/create-sitecore-jss/src/init-runner.ts [hunk 6]
```diff
@@ -41,16 +41,19 @@ export const initRunner = async (initializers: string[], args: BaseArgs) => {
...
+    // final steps (install, lint, etc)
+    if (!args.noInstall) {
+      installPackages(destination, args.silent);
+      lintFix(destination, args.silent);
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/init-runner.ts:50]: Our proxy apps don't have linting configured. Skip this for those?

File: packages/create-sitecore-jss/src/init-runner.ts [hunk 7]
```diff
@@ -41,16 +41,19 @@ export const initRunner = async (initializers: string[], args: BaseArgs) => {
...
+    }
 
-  // install pre-push hook if user opts-in
-  if (args.prePushHook) {
-    await installPrePushHook(args.destination, args.silent);
+    // install pre-push hook if user opts-in
+    if (args.prePushHook) {
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/init-runner.ts:54]: Same here (proxy apps don't include pre-push hooks). Skip this for those?

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: packages/create-sitecore-jss/src/bin.ts (hunk 3 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/bin.ts:20]: `proxyDestination` reads like a request destination. Let me use `proxyAppDestination` then

File: packages/create-sitecore-jss/src/init-runner.ts (hunk 6 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/init-runner.ts:50]: Do we intend it to not have lint, ever?

Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/init-runner.ts:50]: I think it's better to call this and other util functions regardless and let the create logic check whether the commands are present or not in package.json. This is simpler, more intuitive and allows us worry less if we decide to add lint and other things to proxies later on.
Prior to this PR the functions would be launched anyway when create-sitecore-jss is called for proxy only.

Comment (ambrauer) [reply packages/create-sitecore-jss/src/init-runner.ts:50]: Ah, you're right. I agree, makes sense to leave it then.

File: packages/create-sitecore-jss/src/init-runner.ts (hunk 7 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/init-runner.ts:55]: Following on my comment above, I added a check to see if install-hook command exists.
I believe the scaffold logic should be flexible and more universal with its flags and logic behind them

**Review by ambrauer — APPROVED** [review]
👍 Looks good!

**Pull request merged**